### ISSUE TITLE
Arbitrary Account Deletion - Insecure Direct Object Reference

### ISSUE THREAD

**Issue by iamaldi** [TsimpDim/3RStore#89]
## Overview
This is a critical issue, it basically allows a user to delete any other account on the application.
Prerequisites are that a user is logged in.
All a user has to do is to modify the `user_id` parameter of the request that hits the `/delacc` endpoint

## The Bug
The responsible code can be found below:
https://github.com/TsimpDim/3RStore/blob/49ccb10f49cc7aee6d825aa28484ea9a9475f11e/_3RStore/views.py#L131-L151

Basically the application extracts the `user_id` from the untrusted user input of the form of the HTTP request.
## Remediation Advice
The application should extract that information from the currently logged in user's session
`session['user_id']` and not trust the form data which in the end, can be manipulated.
